sdk/core/azure-core/test/ut/nullable_test.cpp

```diff
@@ -224,3 +224,36 @@ TEST(Nullable, Move)
   EXPECT_EQ(*taken, 123);
   // val.HasValue() would return true, but accessing a value after it has been moved is UB anyways.
 }
+
+TEST(Nullable, ConstexprAndRvalue)
+{
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpessimizing-move" // cspell:disable-line
+#endif // __clang__
+
+  Nullable<int> nullableInt0(std::move(Nullable<int>()));
+  Nullable<int> nullableInt11(std::move(Nullable<int>(11)));
+
+#if defined(__clang__)
+#pragma clang diagnostic pop // NOLINT(clang-diagnostic-unknown-pragmas)
+#endif // __clang__
```

Comment:
Why is this pragma needed when doing the `std::move` on `Nullable<T>`?

Also, what are we trying to test here exactly?

Reply:
It is testing move constructor, and suppressing the warnings that the code written makes little sense.